Mathlib/Data/Fin/Basic.lean

```diff
@@ -510,6 +521,11 @@ theorem val_one'' {n : ℕ} : ((1 : Fin (n + 1)) : ℕ) = 1 % (n + 1) :=
   rfl
 #align fin.one_val Fin.val_one''
 
+theorem castSucc_one' [NeZero n] : castSucc (1 : Fin (n + 1)) = 1 := by
+  rw [ext_iff, coe_castSucc, val_one', val_one, mod_succ_eq_iff_lt, succ_eq_one_add,
+  lt_add_iff_pos_right]
+  exact NeZero.pos n
```

Comment:
Note that castSucc_one' exists, with the NeZero assumption. The answer to this is "because Std doesn't have NeZero", I guess. Technically you could get rid of all the n+2 lemmas, but the convention seems to generally be to have both. It might be that's just a limitation of the fact that Std has a lot of these though.

It is never correct in my view to *only* have the n+2 case, you really want to avoid having to ever eliminate the trivial case, even though this can often be annoying. For instance, predAbove, because it takes in a Fin n as an argument, could really be defined for n+1 only - but you don't need this to be true, and if you can do it in generality, you should.